packages/compass-aggregations/src/components/aggregation-side-panel/stage-wizard-use-cases/lookup/lookup.tsx

```diff
@@ -0,0 +1,190 @@
+import {
+  Body,
+  spacing,
+  css,
+  TextInput,
+  Combobox,
+  ComboboxWithCustomOption,
+  ComboboxOption,
+} from '@mongodb-js/compass-components';
+import React, { useCallback, useMemo, useState } from 'react';
+import { connect } from 'react-redux';
+import type { RootState } from '../../../../modules';
+import { fetchCollectionFields } from '../../../../modules/collections-fields';
+import type { CollectionData } from '../../../../modules/collections-fields';
+
+const LOOKUP_TITLE = 'Join documents in';
+
+type LookupFormState = {
+  from: string;
+  localField: string;
+  foreignField: string;
+  as: string;
+};
+
+const containerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[2],
+});
+
+const titleStyles = css({
+  minWidth: `${LOOKUP_TITLE.length}ch`,
+  textAlign: 'right',
+});
+
+const formGroup = css({
+  display: 'flex',
+  gap: spacing[2],
+  alignItems: 'center',
+});
+
+const inputFieldStyles = css({
+  width: '300px',
+});
+
+export const LookupForm = ({
+  fields,
+  collectionsFields,
+  onSelectCollection,
+  onChange,
+}: {
+  fields: string[];
+  collectionsFields: Record<string, CollectionData>;
+  onSelectCollection: (collection: string) => void;
+  onChange: (value: string, error: Error | null) => void;
+}) => {
+  const [formData, setFormData] = useState<LookupFormState>({
+    as: '',
+    from: '',
+    foreignField: '',
+    localField: '',
+  });
+
+  const onSelectOption = useCallback(
+    (option: keyof LookupFormState, value: string | null) => {
+      if (value === null) {
+        return;
+      }
+
+      // We set the "as" form field if:
+      // 1. The value was not set initially and
+      // 2. The value was set initially programmatically and
+      //  was not changed by the user or the value is empty
+      const newData: LookupFormState = { ...formData, [option]: value };
+      if (
+        option === 'from' &&
+        (!formData.as || formData.from === formData.as)
+      ) {
+        newData.as = value;
+      }
+
+      // Call onChange with the new value and an error if any of the fields is empty
+      const anyEmptyValue = Object.values(newData).some((x) => !x);
+      onChange(
+        JSON.stringify(newData),
+        anyEmptyValue ? new Error('Enter all the fields') : null
+      );
+
+      // Fetch the fields of the selected collection if it was changed
+      if (option === 'from' && newData.from) {
```

Comment:
Would this function get triggered if the user selects the same `from` option (the one that is already selected)? If yes, then we probably should apply the suggestion:

```suggestion
      if (option === 'from' && newData.from !== formData.from) {
```

Reply:
added